test/framework/clustervalidator.go

```diff
@@ -3,277 +3,104 @@ package framework
+	csvalidations "github.com/aws/eks-anywhere/test/framework/clusterstatevalidations"
 )
 
-// TODO: Move ClusterValidator to a separate package
-
-// ClusterValidation defines a validation that can be registered to the ClusterValidator.
-type ClusterValidation = func(ctx context.Context, vc ClusterValidatorConfig) error
-
-type retriableValidation struct {
-	validation    ClusterValidation // the validation to run against the cluster.
-	backoffPeriod time.Duration     // the duration to wait another validation attempt.
-	maxRetries    int               // the maximum number of retries to validate.
-}
-
-// ClusterValidator is responsible for checking if a cluster is valid against the spec that is provided.
-type ClusterValidator struct {
-	Config      ClusterValidatorConfig
-	validations []retriableValidation
-}
-
-// WithValidation registers a validation to the ClusterValidator that will be run when Validate is called.
-func (c *ClusterValidator) WithValidation(validation ClusterValidation, backoffPeriod time.Duration, maxRetries int) {
-	c.validations = append(c.validations, retriableValidation{validation, backoffPeriod, maxRetries})
-}
-
-// WithValidations registers multiple validations to the ClusterValidator that will be run when Validate is called.
-func (c *ClusterValidator) WithValidations(validation ...ClusterValidation) {
-	for _, v := range validation {
-		c.validations = append(c.validations, retriableValidation{v, 5 * time.Second, 30})
-	}
-}
-
-// WithWorkloadClusterValidations registers a validation set for a workload cluster.
-func (c *ClusterValidator) WithWorkloadClusterValidations() {}
-
-// WithExpectedObjectsExist registers a set of validations for the existence of various cluster objects.
-func (c *ClusterValidator) WithExpectedObjectsExist() {
-	c.WithValidation(validateClusterReady, 5*time.Second, 60)
-	c.WithValidation(validateEKSAObjects, 5*time.Second, 60)
-	c.WithValidation(validateControlPlaneNodes, 5*time.Second, 120)
-	c.WithValidation(validateWorkerNodes, 5*time.Second, 120)
-	c.WithValidation(validateCilium, 5*time.Second, 60)
-}
-
-// WithClusterDoesNotExist registers a validation to check that a cluster does not exist or has been deleted.
-func (c *ClusterValidator) WithClusterDoesNotExist() {
-	c.WithValidation(validateClusterDoesNotExist, 5*time.Second, 60)
-}
-
-// Validate runs through the set registered validations and returns an error if any of them fail after a number of retries.
-func (c *ClusterValidator) Validate(ctx context.Context) error {
-	for _, v := range c.validations {
-		err := retrier.Retry(v.maxRetries, v.backoffPeriod, func() error {
-			return v.validation(ctx, c.Config)
-		})
-		if err != nil {
-			return fmt.Errorf("validation failed %v", err)
-		}
+// WithValidationsForExpectedObjects registers a set validations for the existence of various cluster objects.
+func WithValidationsForExpectedObjects(cv *clusterf.StateValidator) *clusterf.StateValidator {
```

Comment:
nit: I would return a validation slice instead of doing the registering here
That way you have all the register calls in one place and avoid methods with side effects